<a href="https://colab.research.google.com/github/Frexxza26/BelajarRepositoryMIKTI/blob/Frexxza26-colabs-20240304/EDA_Armanda_Fathurrahman.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Nama: Armanda Fathurrahman

#NIM: 09011282126055

#Kelas: SK6A Indralaya

###**ANALISIS EKSPLORASI DATA**

Analisis Eksplorasi Data (Exploratory Data Analysis atau EDA) merupakan suatu pendekatan dalam statistika dan ilmu data yang bertujuan untuk memahami struktur dan karakteristik dari dataset yang dimiliki. EDA melibatkan serangkaian teknik analisis statistik dan visualisasi data untuk mengidentifikasi pola, tren, anomali, serta mendapatkan wawasan awal terhadap data tanpa membuat asumsi awal yang kuat. Tujuan utama EDA adalah untuk membantu dalam merumuskan pertanyaan-pertanyaan yang lebih spesifik dan merinci, serta mengidentifikasi potensi hipotesis untuk diuji lebih lanjut.

Tahap ini memiliki beberapa aspek yang penting dalam proses pengolahan data selanjutnya. Aspek-aspek tersebut ialah sebagai berikut:



1.   Perincian data secara statistik deskriptif.
2.   Visualisasi data.
3.   Pembersihan data terhadap anomali.
4.   Handling Missing Values



## Proses *EDA*

Analisis Data Eksplorasi kali ini akan menggunakan dataset kanker payudara. Sebelum memulai langkah pertama, saya akan melakukan integrasi Google Colab dengan Google Drive. Hal ini memungkinkan saya untuk dapat mengakses data yang akan saya gunakan.

In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


Selanjutnya, saya menetapkan direktori file data saya, dengan tujuan untuk mendeteksi file yang ada di dalamnya, serta menetapkan folder tersebut sebagai folder utama

In [ ]:
import os
print(os.listdir('/content/gdrive/My Drive/Input'))

['breast cancer.csv']


Ini bertujuan mendeteksi ada file apa saja di dalam folder Input. Tertulis bahwa *breast cancer.csv* terdeteksi disana. Artinya, kita dapat lanjut eksplorasi data tersebut.

In [ ]:
import pandas as pd
import numpy as np
from matplotlib import pyplot as plt
import seaborn as sns

Import beberapa library, salah satunya adalah library Pandas untuk memudahkan dalam pengolahan data.

In [ ]:
cancer = pd.read_csv('/content/gdrive/My Drive/Input/breast cancer.csv')

Lalu, lakukan import data csv ke variabel "cancer"

In [ ]:
cancer.head()

id diagnosis  radius_mean  texture_mean  perimeter_mean  area_mean  \
0    842302         M        17.99         10.38          122.80     1001.0   
1    842517         M        20.57         17.77          132.90     1326.0   
2  84300903         M        19.69         21.25          130.00     1203.0   
3  84348301         M        11.42         20.38           77.58      386.1   
4  84358402         M        20.29         14.34          135.10     1297.0   

   smoothness_mean  compactness_mean  concavity_mean  concave points_mean  \
0          0.11840           0.27760          0.3001              0.14710   
1          0.08474           0.07864          0.0869              0.07017   
2          0.10960           0.15990          0.1974              0.12790   
3          0.14250           0.28390          0.2414              0.10520   
4          0.10030           0.13280          0.1980              0.10430   

   ...  texture_worst  perimeter_worst  area_worst  smoothness_worst  \
0  ...          17.33           184.60      2019.0            0.1622   
1  ...          23.41           158.80      1956.0            0.1238   
2  ...          25.53           152.50      1709.0            0.1444   
3  ...          26.50            98.87       567.7            0.2098   
4  ...          16.67           152.20      1575.0            0.1374   

   compactness_worst  concavity_worst  concave points_worst  symmetry_worst  \
0             0.6656           0.7119                0.2654          0.4601   
1             0.1866           0.2416                0.1860          0.2750   
2             0.4245           0.4504                0.2430          0.3613   
3             0.8663           0.6869                0.2575          0.6638   
4             0.2050           0.4000                0.1625          0.2364   

   fractal_dimension_worst  Unnamed: 32  
0                  0.11890          NaN  
1                  0.08902          NaN  
2                  0.08758          NaN  
3                  0.17300          NaN  
4                  0.07678          NaN  

[5 rows x 33 columns]

Berikut merupakan 5 data pertama pada dataset breast cancer.csv. Terlihat bahwa ada 33 fitur yang tersedia pada dataset ini. Berikut penjelasan dari tiap fiturnya.



1.   **ID** : Merupakan nomor id pada sel kanker per barisnya.
2.   **Diagnosis** : Merupakan jenis sel kanker. Ada 2 jenis sel kanker, yakni Ganas (*Malignant (M)*) dan Jinak (*Benign (B)*).
3.   **radius_mean** : Merupakan radius rata-rata dari setiap sel kanker.
4.   **texture_mean** : Merupakan tekstur rata-rata dari setiap sel kanker.
5.   **perimeter_mean** : Merupakan keliling rata-rata dari setiap sel kanker.
6.   **area_mean** : Merupakan luas rata-rata dari setiap sel kanker.
7.   **smoothness_mean** : Merupakan tingkat kehalusan rata-rata dari setiap sel kanker.
8.   **compactness_mean** : Merupakan kepadatan rata-rata dari setiap sel kanker.
9.   **concavity_mean** : Merupakan kecekungan rata-rata dari setiap sel kanker.
10.  **concavepoints_mean** : Merupakan titik cekung rata-rata dari setiap sel kanker.
11.  **symmetry_mean** : Merupakan rata-rata simetri dari setiap sel.
12.  **fractaldimension_mean** : Merupakan dimensi fraktal rata-rata dari setiap sel kanker.
13.  **radius_se** : Merupakan banyaknya variasi radius dari setiap sel kanker.
14.  **texture_se** : Merupakan banyaknya variasi tekstur dari setiap sel kanker.
15.  **perimeter_se** : Merupakan banyaknya variasi keliling dari setiap sel kanker.
16.  **area_se** : Merupakan banyaknya variasi luas dari setiap sel kanker.
17.  **smoothness_se** : Merupakan banyaknya variasi tingkat kehalusan dari setiap sel kanker.
18.  **compactness_se** : Merupakan banyaknya variasi kepadatan dari setiap sel kanker.
19.  **concavity_se** : Merupakan banyaknya variasi kecekungan dari setiap sel kanker.
20.  **concavepoints_se** : Merupakan banyaknya variasi titik cekung dari setiap sel kanker.
21.  **symmetry_se** : Merupakan banyaknya variasi simetri dari setiap sel.
22.  **fractaldimension_se** : Merupakan banyaknya variasi dimensi fraktal dari setiap sel kanker.
23.  **radius_worst** : Merupakan radius terburuk dari setiap sel kanker.
24.  **texture_worst** : Merupakan tekstur terburuk dari setiap sel kanker.
25.  **perimeter_worst** : Merupakan keliling terburuk dari setiap sel kanker.
26.  **area_worst** : Merupakan luas terburuk dari setiap sel kanker.
27.  **smoothness_worst** : Merupakan tingkat kehalusan terburuk dari setiap sel kanker.
28.  **compactness_worst** : Merupakan kepadatan terburuk dari setiap sel kanker.
29.  **concavity_worst** : Merupakan kecekungan terburuk dari setiap sel kanker.
30.  **concavepoints_worst** : Merupakan titik cekung terburuk dari setiap sel kanker.
31.  **symmetry_worst** : Merupakan terburuk simetri dari setiap sel.
32.  **fractaldimension_worst** : Merupakan dimensi fraktal terburuk dari setiap sel kanker.
33.  **Unnamed:32** : Tidak termasuk sebagai fitur untuk diolah.

Catatan : Standard Error (SE) mengacu pada seberapa jauh rata-rata dari suatu sampel dapat diharapkan untuk mendekati rata-rata sebenarnya dari populasi.

In [ ]:
cancer.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 569 entries, 0 to 568
Data columns (total 33 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   id                       569 non-null    int64  
 1   diagnosis                569 non-null    object 
 2   radius_mean              569 non-null    float64
 3   texture_mean             569 non-null    float64
 4   perimeter_mean           569 non-null    float64
 5   area_mean                569 non-null    float64
 6   smoothness_mean          569 non-null    float64
 7   compactness_mean         569 non-null    float64
 8   concavity_mean           569 non-null    float64
 9   concave points_mean      569 non-null    float64
 10  symmetry_mean            569 non-null    float64
 11  fractal_dimension_mean   569 non-null    float64
 12  radius_se                569 non-null    float64
 13  texture_se               569 non-null    float64
 14  perimeter_se             5

Berikut merupakan tipe data dari dataset yang digunakan. Informasi yang dapat ditarik adalah sebagai berikut.

1.   Terdapat 1 data kategorikal, yakni *diagnosis* yang berbentuk **object**
2.   Terdapat 32 data numerik, yakni *id* dengan bentuk **int64**, dan sisanya berbentuk **float64**

Berikut merupakan tipe data dari dataset setelah dilakukan proses penghapusan fitur *id* dan *Unnamed : 32*. Informasi yang dapat ditarik adalah sebagai berikut.



1.   Terdapat 1 data kategorikal, yakni *diagnosis* yang berbentuk **object**
2.   Terdapat 30 data numerik yang berbentuk **float64**



In [ ]:
cancer.isnull().sum()

id                           0
diagnosis                    0
radius_mean                  0
texture_mean                 0
perimeter_mean               0
area_mean                    0
smoothness_mean              0
compactness_mean             0
concavity_mean               0
concave points_mean          0
symmetry_mean                0
fractal_dimension_mean       0
radius_se                    0
texture_se                   0
perimeter_se                 0
area_se                      0
smoothness_se                0
compactness_se               0
concavity_se                 0
concave points_se            0
symmetry_se                  0
fractal_dimension_se         0
radius_worst                 0
texture_worst                0
perimeter_worst              0
area_worst                   0
smoothness_worst             0
compactness_worst            0
concavity_worst              0
concave points_worst         0
symmetry_worst               0
fractal_dimension_worst      0
Unnamed:

Tujuan melakukan *isnan* adalah mengecek, apakah terdapat value yang NaN pada dataset, yang nantinya akan mengganggu dalam proses pengolahan selanjutnya. Dapat dilihat, bahwa data pada tiap fitur tidak memiliki nilai NaN, artinya kita tidak perlu menggunakan sintaks **fillna**.

In [ ]:
cancer['diagnosis'].value_counts()

B    357
M    212
Name: diagnosis, dtype: int64

Fitur *diagnosis* merupakan data kategorikal, yang mana pada data ini, terdapat 2 kategori sel kanker, yakni Ganas dan Jinak. Terdapat 357 sel kanker jinak, dan 212 sel kanker ganas.

In [ ]:
cancer.describe().T

count          mean           std          min  \
id                       569.0  3.037183e+07  1.250206e+08  8670.000000   
radius_mean              569.0  1.412729e+01  3.524049e+00     6.981000   
texture_mean             569.0  1.928965e+01  4.301036e+00     9.710000   
perimeter_mean           569.0  9.196903e+01  2.429898e+01    43.790000   
area_mean                569.0  6.548891e+02  3.519141e+02   143.500000   
smoothness_mean          569.0  9.636028e-02  1.406413e-02     0.052630   
compactness_mean         569.0  1.043410e-01  5.281276e-02     0.019380   
concavity_mean           569.0  8.879932e-02  7.971981e-02     0.000000   
concave points_mean      569.0  4.891915e-02  3.880284e-02     0.000000   
symmetry_mean            569.0  1.811619e-01  2.741428e-02     0.106000   
fractal_dimension_mean   569.0  6.279761e-02  7.060363e-03     0.049960   
radius_se                569.0  4.051721e-01  2.773127e-01     0.111500   
texture_se               569.0  1.216853e+00  5.516484e-01     0.360200   
perimeter_se             569.0  2.866059e+00  2.021855e+00     0.757000   
area_se                  569.0  4.033708e+01  4.549101e+01     6.802000   
smoothness_se            569.0  7.040979e-03  3.002518e-03     0.001713   
compactness_se           569.0  2.547814e-02  1.790818e-02     0.002252   
concavity_se             569.0  3.189372e-02  3.018606e-02     0.000000   
concave points_se        569.0  1.179614e-02  6.170285e-03     0.000000   
symmetry_se              569.0  2.054230e-02  8.266372e-03     0.007882   
fractal_dimension_se     569.0  3.794904e-03  2.646071e-03     0.000895   
radius_worst             569.0  1.626919e+01  4.833242e+00     7.930000   
texture_worst            569.0  2.567722e+01  6.146258e+00    12.020000   
perimeter_worst          569.0  1.072612e+02  3.360254e+01    50.410000   
area_worst               569.0  8.805831e+02  5.693570e+02   185.200000   
smoothness_worst         569.0  1.323686e-01  2.283243e-02     0.071170   
compactness_worst        569.0  2.542650e-01  1.573365e-01     0.027290   
concavity_worst          569.0  2.721885e-01  2.086243e-01     0.000000   
concave points_worst     569.0  1.146062e-01  6.573234e-02     0.000000   
symmetry_worst           569.0  2.900756e-01  6.186747e-02     0.156500   
fractal_dimension_worst  569.0  8.394582e-02  1.806127e-02     0.055040   
Unnamed: 32                0.0           NaN           NaN          NaN   

                                   25%            50%           75%  \
id                       869218.000000  906024.000000  8.813129e+06   
radius_mean                  11.700000      13.370000  1.578000e+01   
texture_mean                 16.170000      18.840000  2.180000e+01   
perimeter_mean               75.170000      86.240000  1.041000e+02   
area_mean                   420.300000     551.100000  7.827000e+02   
smoothness_mean               0.086370       0.095870  1.053000e-01   
compactness_mean              0.064920       0.092630  1.304000e-01   
concavity_mean                0.029560       0.061540  1.307000e-01   
concave points_mean           0.020310       0.033500  7.400000e-02   
symmetry_mean                 0.161900       0.179200  1.957000e-01   
fractal_dimension_mean        0.057700       0.061540  6.612000e-02   
radius_se                     0.232400       0.324200  4.789000e-01   
texture_se                    0.833900       1.108000  1.474000e+00   
perimeter_se                  1.606000       2.287000  3.357000e+00   
area_se                      17.850000      24.530000  4.519000e+01   
smoothness_se                 0.005169       0.006380  8.146000e-03   
compactness_se                0.013080       0.020450  3.245000e-02   
concavity_se                  0.015090       0.025890  4.205000e-02   
concave points_se             0.007638       0.010930  1.471000e-02   
symmetry_se                   0.015160       0.018730  2.348000e-02   
fractal_dimension_se          0.002248       0.003187  4.5

Berikut merupakan penjabaran terkait informasi umum tiap fitur yang ada pada dataset. Berikut adalah penjelasannya.

1. **Count**: Adalah jumlah baris yang ada pada dataset. Terdapat 569 baris pada dataset ini.

2. **Mean**: Menjelaskan terkait nilai rata-rata yang dimiliki suatu fitur. Caranya, dengan menambahkan seluruh nilai pada masing-masing fitur akan dijumlahkan, kemudian dikurangi dengan total baris dataset. Contohnya, pada fitur radius_mean, dijumlahkan seluruhnya, didapat hasilnya 8038.429. Kemudian, hasilnya akan dibagi dengan jumlah baris, yakni 569. Sehingga didapat

> 8038.429 / 569 = 14.12729173989455

Artinya, nilai rata-rata dari fitur radius_mean adalah 14.12729173989455, yang bisa disederhanakan menjadi 1.412729e+01. Hal ini juga berlaku dengan fitur-fitur lainnya.

3. **std**: Std atau standar deviasi adalah seberapa tersebar atau seberapa jauh nilai-nilai dalam suatu dataset dari rata-rata (mean) dataset tersebut. Standar deviasi memberikan gambaran tentang sejauh mana titik data dari rata-rata, dan semakin tinggi standar deviasi, semakin besar variasi atau dispersi data.

Ada dua hal yang patut diperhatikan pada standar deviasi ini, yaitu:
*   Jika standar deviasi mendekati nol, maka sebagian besar nilai dalam dataset cenderung mendekati rata-rata.
*   jika standar deviasi besar dan tidak mendekati nol, maka nilai-nilai dalam dataset tersebar jauh dari rata-rata, dan dataset memiliki variasi yang tinggi.

Contohnya kita ambil radius_mean dan area_mean. Bila kita lihat berdasarkan tabel diatas, dapat disimpulkan bahwa radius_mean memiliki persebaran nilai yang lebih mendekati rata-rata dengan nilai **3.524049e+00** dibandingkan area_mean yang memiliki nilai **3.519141e+02**.

4. **min**: Menampilkan nilai minimum atau nilai terendah yang ada pada tiap fitur.

5. **25%**: Menampilkan nilai kuartil pertama yang ada di antara nilai min dan mean.

6. **50%**: Menampilkan nilai kuartil kedua (median) yang terletak di tengah-tengah data.

7. **75%**: Menampilkan nilai kuartil ketiga yang ada di antara median dan nilai maksimal suatu data.

8. **Max** = Menampilkan nilai maksimum yang ada pada tiap fiturnya.